### PR TITLE
On Endpoints changes, only queue reconciliation of ingress if it's related to the changed Endpoints object

### PR DESCRIPTION
**Pull request by devkid**
@OmerKahani I saw your pull request and we had the same problem, so I implemented the missing functionality.

Closes #1173.

### CODE REVIEW

File: internal/ingress/controller/handlers/endpoints.go [hunk 1]
```diff
@@ -48,7 +51,6 @@ func (h *EnqueueRequestsForEndpointsEvent) Delete(e event.DeleteEvent, queue wor
 func (h *EnqueueRequestsForEndpointsEvent) Generic(event.GenericEvent, workqueue.RateLimitingInterface) {
 }
 
-//TODO: this can be further optimized to only included ingresses referenced this endpoints(service) :D
```
Comment (M00nF1sh) [inline internal/ingress/controller/handlers/endpoints.go]: nice 🤣

File: internal/ingress/controller/handlers/endpoints.go [hunk 2]
```diff
@@ -68,3 +70,41 @@ func (h *EnqueueRequestsForEndpointsEvent) enqueueImpactedIngresses(endpoints *c
...
+				}
+			}
+		}
+	}
+	return false
+}
+
+// EndpointsBelongsToBackend checks if the given endpoints object is associated with the given backend of the given ingress
+func EndpointsBelongsToBackend(endpoints *corev1.Endpoints, ingress *extensions.Ingress, backend *extensions.IngressBackend) bool {
+	if backend.ServiceName == "use-annotation" {
```
Comment (M00nF1sh) [inline internal/ingress/controller/handlers/endpoints.go]: should be `backend.ServicePort`

**Comment by OmerKahani** [comment]
Hi, I had of a diffrent implementation in mind, and I am still in the process of writting the tests for it

**Comment by M00nF1sh** [comment]
> Hi, I had of a diffrent implementation in mind, and I am still in the process of writting the tests for it

waits for your implementation and see which one is better 🤣

**Comment by M00nF1sh** [comment]
closing this in favor of https://github.com/kubernetes-sigs/aws-alb-ingress-controller/pull/1179

**Pull request closed without merging**